Add EventListener details page

a-roberts opened the pull request:
For https://github.com/tektoncd/dashboard/issues/671

a-roberts commented:
![image](https://user-images.githubusercontent.com/12814972/71019694-b8a1b000-20f2-11ea-92bc-17b4fe0df275.png)

this isn't looking pretty so far...

![image](https://user-images.githubusercontent.com/12814972/71019717-c0f9eb00-20f2-11ea-8602-96f6f03b7cce.png)

I have oddities in the table (especially on hover and around the headers)

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,167 @@
+          defaultMessage: 'TriggerTemplate'
+        })}
+      </h5>
+      <Button
```
We can add the LinkIcon into the Link content below instead of using a button here
same for the binding above

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,167 @@
+
+  let interceptorValues = [];
+
+  if (trigger.interceptor.header) {
```
`interceptor` is optional so may not be present

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,167 @@
+
+  return (
+    <>
+      <h5>
```
avoid using headings non-semantically, looks like we're skipping h2 - h4 here
we can re-style them if needed

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,167 @@
+      <h5>
+        {intl.formatMessage({
+          id: 'dashboard.triggerDetails.triggerBinding',
+          defaultMessage: 'TriggerBinding'
```
no need to translate the resource kinds

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,167 @@
+      >
+        {trigger.template.name}
+      </Link>
+      <br /> <br />
```
use margins instead of `<br />` for spacing

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,167 @@
+          id: 'dashboard.trigger.noParams',
+          defaultMessage: 'No parameters found for this Trigger.'
+        })}
+        isSortable
```
Carbon doesn't seem to like nested Tables, especially when sorting is enabled. Removing `isSortable` resolves the weird table header layout. The hover styles still need work but at least the headers are consistent.

AlanGreene commented on packages/components/src/components/Trigger/Trigger.scss:
```diff
@@ -0,0 +1,24 @@
+limitations under the License.
+*/
+
+pre {
```
we should probably move this to the YAML component

AlanGreene commented on src/containers/EventListener/EventListener.js:
```diff
@@ -0,0 +1,269 @@
+    }
+
+    /* eslint-disable prefer-destructuring */
+    const triggers = eventListener.spec.triggers;
```
no need for the lint override
`const { triggers } = eventListener.spec;`

AlanGreene commented on src/containers/EventListener/EventListener.js:
```diff
@@ -0,0 +1,269 @@
+          </Tab>
+        </Tabs>
+
+        <div className="tableComponent">
```
should this content be inside the details tab?

a-roberts commented:
@AlanGreene please take another look, resolved most of the issues apart from the LinkIcon part, stuck on how to do this 🤔

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,150 @@
+            triggerBindingName: trigger.binding.name
+          })}
+        >
+          <span className="truncate-text-end">{trigger.binding.name}</span>
```
should have `title={trigger.binding.name}` on the span to ensure the user can still view the full name on hover when truncated

a-roberts commented:
@AlanGreene thanks Alan for all of your feedback, added the titles 😄 

I'd still like to turn off the flashing behaviour of the data table on hover but cool with doing this after the Christmas break

@carlos-logro FYI

a-roberts commented:
@AlanGreene ready for another look please, still can't figure out the on-hover behaviour though, I think it's because it's inside of an expandable row (so when the user hovers over an expanded row, we lose all of the nice zebra stripes)

Changes since last time include removing the v1 text as it was pointless really, I've also made the table less grey and used `Headers` instead of `Interceptor headers`. Also added some padding.

Screenshots:
![image](https://user-images.githubusercontent.com/12814972/71811074-bab0bc00-306b-11ea-8c23-36e898a498a1.png)

![image](https://user-images.githubusercontent.com/12814972/71811090-c2706080-306b-11ea-9656-70d6415f7662.png)

AlanGreene commented on packages/components/src/components/Trigger/Trigger.js:
```diff
@@ -0,0 +1,154 @@
+          })}
+        </h2>
+
+        <Table // If anyone knows how to make this not go all grey on-hover please fix
```
Looks like Carbon doesn't want to support or encourage use of nested DataTables: https://github.com/carbon-design-system/carbon-components-react/pull/2108#issuecomment-477664867
However, you may find this useful: https://github.com/carbon-design-system/carbon/pull/2222 (for Carbon 9 but might still be applicable)

AlanGreene commented on packages/components/src/components/Trigger/Trigger.scss:
```diff
@@ -0,0 +1,74 @@
+
+@import '~carbon-components/scss/globals/scss/vars';
+
+.bx--btn.link-btn.bx--btn--ghost {
```
do we want to apply this globally or only in specific contexts?

a-roberts replied:
I reckon globally, how? I know we've got a vars file but it's not going to be a dollared variable is it?

AlanGreene replied:
styles in App.scss apply to all pages, so we could move it there

a-roberts commented:
Thanks @AlanGreene, on it!

a-roberts commented:
Dropped the nested tables 👀

AlanGreene commented on src/containers/EventListener/EventListener.js:
```diff
@@ -0,0 +1,217 @@
+              {triggers.map(theTrigger => {
+                return (
+                  <Trigger
+                    eventListenerNamespace={eventListenerNamespace}
```
missing `key` prop

a-roberts commented:
Updated
